File: intro_to_business_analytics/i2ba_chap7.py
import polars as pl
from datetime import datetime

delivDF = pl.read_csv('shipDF.csv', try_parse_dates=True)
delivDF

shipDF = (
    delivDF
    .with_columns(
        pl.col("plannedShipDate").str.to_datetime("%m/%d/%Y"),
        pl.col("actualShipDate").str.to_datetime("%m/%d/%Y")
    )
)
shipDF

shipDF["actualShipDate"][0] - shipDF["plannedShipDate"][0]

thisDay = datetime.now()

thisDay.month
thisDay.year
thisDay.weekday()

shipDateDF = (
    shipDF
    .select("shipID", "plannedShipDate", "actualShipDate")
    .unique()
    .sort("shipID")
    .with_columns(
        lateFlag = pl.col("actualShipDate") > pl.col("plannedShipDate")
    )
)
shipDateDF

(
    shipDateDF
    .group_by("lateFlag")
    .agg(countLate = pl.col("shipID").len())
    .with_columns(proportion = pl.col("countLate") / pl.col("countLate").sum())
)

# 7.3

File: intro_to_business_analytics/i2ba_chap7.R
library("causact")
library("lubridate")
library("dplyr")
library("readr")

data("delivDF")

shipDF <- delivDF

write_csv(shipDF, 'shipDF.csv')

shipDF$plannedShipDate <- mdy(shipDF$plannedShipDate)
shipDF$actualShipDate <- mdy(shipDF$actualShipDate)

shipDF$actualShipDate[1] - shipDF$plannedShipDate[1]

today()

thisDay = today()
year(thisDay)
month(thisDay)
wday(thisDay)
yday(thisDay)

wday(thisDay, label = TRUE)

shipDateDF <- shipDF |> 
  select(shipID, plannedShipDate, actualShipDate) |> 
  distinct() |> 
  mutate(
    lateFlag = ifelse(actualShipDate > plannedShipDate, TRUE, FALSE)
  )

shipDateDF |> 
  group_by(lateFlag) |> 
  summarise(countLate = n()) |> 
  mutate(proportion = countLate / sum(countLate))

data(prodLineDF)
prodLineDF

write_csv(prodLineDF, 'prodLineDF.csv')

catDF <- shipDF |> 
  left_join((prodLineDF)) |> 
  filter(!is.na(prodCategory))

# 7.4